tests/modules/umitools/dedup/main.nf

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { UMITOOLS_EXTRACT } from '../../../../modules/umitools/extract/main.nf'
+include { BWA_INDEX } from '../../../../modules/bwa/index/main.nf'
+include { BWA_MEM   } from '../../../../modules/bwa/mem/main.nf'
+include { SAMTOOLS_INDEX   } from '../../../../modules/samtools/index/main.nf'
+include { UMITOOLS_DEDUP } from '../../../../modules/umitools/dedup/main.nf'
+
+//
+// Test with single-end data
+//
+workflow test_umitools_dedup_single_end {
+    input = [ [ id:'test', single_end:true ], // meta map
+              [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) ]
```

Comment:
Could we just use the test bams and bais for this? Or are they already deduped (I don't know exactly what the dedup is doing)? If so, could we add a comment about why the extra steps are included?

Maybe we could keep these to just dedup. Maybe we can keep the e2e tests(which are a good example of how to use the modules), but the minimal tests remove the dependencies on BWA and samtools changing under the hood. I can write up some of those if I'm not being clear.

Reply:
But I do like showing how to actually use the module...

Reply:
> The test bams don't have unique molecular identifiers (UMIs) present.


That's what I was guessing!

> But I do like showing how to actually use the module...


 ¿por que no los dos?
 
 Let's just have both tests. Just add the `--ignore-umi` flag